lib/tasks/fix_nil_ethnicity_arrays.rake

```diff
@@ -1,39 +1,50 @@
 # Run this with RAILS_ENV=production bundle exec rake migrations:fix_nil_ethnicity_arrays application_hbx_id="12345"
 # application_hbx_id is not required but can be used for testing/fixing on a single application.
 # By default, rake will run on all ATP-ingested applications in a draft state
+# To run just a report: RAILS_ENV=production bundle exec rake migrations:fix_nil_ethnicity_arrays report_only=true
 namespace :migrations do
   desc "Remove nils from person and applicant ethnicity arrays"
   task :fix_nil_ethnicity_arrays => :environment do |_task, _args|
     # - find draft applications that were ingested via ATP (or using provided hbx_id)
-    # - check and update person ethnicity array
-    # - check and update applicant ethnicity array
+    # - check and update person ethnicity arrays
+    # - check and update applicant ethnicity arrays
 
     def find_person(person_hbx_id)
       Person.find_by(hbx_id: person_hbx_id)
     rescue StandardError => e
-      puts "Error finding person document - hbx_id: #{person_hbx_id}"
+      ["Error finding person document - hbx_id: #{person_hbx_id}"]
```

Comment:
this is not needed. you can revert this change. This method will return nil when there's no person with hbx id.

Reply:
Person.find_by will raise a Mongoid::Errors::DocumentNotFound if it can't find a matching document.  I changed method to use Person.where, which will return nil if no match is found.